### PR TITLE
Add Movable Amethyst Rule

### PR DESCRIPTION
**Pull request by hugeblank**
As discussed in Episode 79 of ilmangos peaceful challenge. 
He said he'd do it, I figured I could beat him to the punch. :)

Context: https://youtu.be/0-fAKpr_J8k?t=1180

### CODE REVIEW

File: src/main/java/carpet/CarpetSettings.java [hunk 1]
```diff
@@ -315,6 +315,9 @@ public String description() {
     )
     public static boolean hopperCounters = false;
 
+    @Rule( desc = "Allows Budding Amethyst blocks to be pushed by pistons", category = FEATURE )
+    public static boolean pushableAmethyst = false;
```
Comment (DragonEggBedrockBreaking) [inline src/main/java/carpet/CarpetSettings.java:319]: Wouldn't `movableAmethyst` be more consistent, considering carpet's `movableBlockEntities` rule?

Comment (hugeblank) [reply src/main/java/carpet/CarpetSettings.java:319]: Agreed, good thinking. I'll get on renaming

**Comment by gnembon** [comment]
Suggest extending to work with silk touch as well. This will give a choice. Since these two mechanics are not colliding suggesting implementing under the same 'true' value.

**Comment by gnembon** [comment]
snappsshot today - will add that manually

**Pull request merged**